File: app/src/main/java/com/alyndroid/pokemon/db/PokemonDao.java
package com.alyndroid.pokemon.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import com.alyndroid.pokemon.model.Pokemon;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface PokemonDao {

    @Insert
    public void insertPokemon(Pokemon pokemon);

    @Query("delete from fav_table where name =:pokemonName")
    public void deletePokemon(String pokemonName);

    @Query("select * from fav_table")
    public LiveData<List<Pokemon>> getPokemons();
}

File: app/src/main/java/com/whiteside/bokemoncourse/db/PokemonDao.kt
package com.whiteside.bokemoncourse.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.whiteside.bokemoncourse.model.Pokemon

@Dao
interface PokemonDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun insertPokemon(pokemon: Pokemon)

    @Query("select * from fav_pokemons")
    suspend fun getPokemons(): List<Pokemon>

    @Query("delete from fav_pokemons where name =:name")
    suspend fun deletePokemon(name: String)
}